fairlearn/postprocessing/_threshold_optimizer.py

```diff
@@ -427,6 +260,167 @@ def _threshold_optimization_equalized_odds(sensitive_features, labels, scores, g
     return predicted_EO_by_sensitive_feature
 
 
+_SUPPORTED_CONSTRAINTS = {
+    DEMOGRAPHIC_PARITY: _threshold_optimization_demographic_parity,
+    EQUALIZED_ODDS: _threshold_optimization_equalized_odds}
+
+
+def _get_soft_predictions(estimator, X):
+    """Returns soft predictions of a classifier using either `predict_proba`
+    or `decision_function` methods.
+    """
+    if hasattr(estimator, "predict_proba"):
+        return estimator.predict_proba(X)
+    elif hasattr(estimator, "decision_function"):
+        return estimator.decision_function(X)
+    else:
+        raise ValueError("{} provides neither a `predict_proba` nor a "
+                            "`decision_function` method.".format(
+                                estimator.__class__.__name__
+                            ))
+
+
+class ThresholdOptimizer(ClassifierMixin, BaseEstimator):
+    """A classifier based on the threshold optimization approach.
+
+    The procedure followed is described in detail in
+    `Hardt et al. (2016) <https://arxiv.org/abs/1610.02413>`_.
+
+    :param unconstrained_predictor: The trained predictor whose output will be
+    post processed
+    :type unconstrained_predictor: A trained predictor
+
+    :param estimator: An untrained estimator that will be trained, and
+        subsequently its output will be post processed
+    :type estimator: An untrained estimator
+
```

Comment:
let's fix "post processed" -> "postprocessed"  in the above two instances (this was decided in a discussion a while back, but did not propagate to the code yet)